bin/check-es-index-field-count.rb

```diff
@@ -0,0 +1,206 @@
+#! /usr/bin/env ruby
+#
+#   check-es-index-field-count
+#
+# DESCRIPTION:
+#   This plugin checks if the number of fields in ES index(es) is approaching limit. ES by default
+#   puts this limit at 1000 fields per index.
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: aws_es_transport
+#
+# USAGE:
+#   This example checks if the number of fields for an index is reaching its limit set in
+#   index.mapping.total_fields.limit. This check takes as paramater the index name or
+#   comma-separated list of indices, and optionally the type to check on. If type is not specified,
+#   all types in index are examined.
+#   You can also specify an optional value for the limit. When omitted, this will default to 1000.
+#
+#   check-es-query-count.rb -h <hostname or ip> -p 9200
+#           -i <index1>,<index2> --types <type_in_index> -w <pct> -c <pct>
+#
+#   If any indices crossing the specified thresholds (warning/critical), beside the appropriate return code
+#   this check will also output a list of indices with the violated percentage for further troubleshooting.
+# NOTES:
+#
+# LICENSE:
+#   CloudCruiser <devops@hpe.com>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'elasticsearch'
+require 'aws_es_transport'
+require 'sensu-plugins-elasticsearch'
+require 'json'
+
+#
+# ES Index Field Count
+#
+class ESIndexFieldCount < Sensu::Plugin::Check::CLI
+  include ElasticsearchCommon
+
+  option :transport,
```

Comment:
I don't see this being used anywhere do we need this? If support is not implemented then we should not include it.